File: quick_sort.m
function y = quick_sort(x);
n=length(x);
x1 = [];
x2 = [];


if n < 2
    y = x
    return
end


for i = 1:n-1
    
    if x(i)<x(n)
        x1 = [x1,x(i)];
    else
        x2 = [x2,x(i)];
    end
end
y = [quick_sort(x1),x(n),quick_sort(x2)];
end

File: QuickSort.R
QuickSort <-function(data){
  # funkce QuickSort() seradi vzestupne vektor ciselnych dat
  # vstup...neserazeny vektor hodnot
  # vystup ...vzestupne serazeny vektor hodnot
  L=length(data)                                    #delka vstupniho vektoru
  if (is.numeric(data)){                            #podminka pro spravny format vstupniho vektoru
    if (L>=2){                                      #podminka pro delku vstupniho vektoru
      pivot=sample(data,1:1)                        #nahodny vyber pivota ze vstupniho vektoru
      data=c(QuickSort(data[data<pivot]), data[data==pivot], QuickSort(data[data>pivot]))  #vytvoreni serazeneho vektoru 
      #mensi nez pivot, pivot, vetsi nez pivot. soucasne je funkce rekurzivne volana pro oba podvektory
      return(data)                                  #vrati serazeny vektor
      }}
  else { print('Vstupem musi byt ciselny vektor')}  #pokud neni vstupem vektor cisel, hlasi chybu
}